### ISSUE TITLE
Not working with Xcode 15, iOS17

### ISSUE THREAD

**Issue by jbhavin** [rive-app/rive-react-native#209]
hello team,

we upgraded our react native version from **0.76.0 to 0.72.7** with latest Xcode15 and it is stop working. please find screenshot for attachment.

![Screenshot 2023-11-17 at 8 37 53 PM](https://github.com/rive-app/rive-react-native/assets/27915787/d87e13af-fc0e-4d56-9288-76ce2f24d249)
![Screenshot 2023-11-17 at 8 37 35 PM](https://github.com/rive-app/rive-react-native/assets/27915787/9af4b60e-009a-4b39-ada1-469e9110b623)

react-native : 0.72.7
react : 18.2.0
rive-react-native : '^6.1.1'
xcode : 15
mac os: Sonama 14.1.1

**Comment by zplata** [comment]
Hi @jbhavin - we should have an update to this package soon here that might assist with the XCode 15 support. We'll comment back here when that release lands and you can try it out.

**Comment by immanjeba98** [comment]
is it update?

**Comment by brascene** [comment]
Same issue here, and when I hardcode the minimum iOS deployment target in Podfile to 14, `pod install` goes well, but app crashes on start even i didn't use the lib anywhere in the code, just installed it, and error is something in react-core, with console message: `Unbalanced calls start/end for tag 19`

<img width="1004" alt="Screenshot 2024-07-17 at 12 26 16" src="https://github.com/user-attachments/assets/41708ac5-89a2-47aa-86f2-e6c251f285db">

I tried this with:

Xcode: 15.4 (15F31d)
iPhone 15 Pro (17.0.1)
React Native: 0.74.2
Rive-React-Native: ^7.3.0

**Comment by JslinSir** [comment]
Add the following

```js
require Pod::Executable.execute_command('node', ['-p',
  'require.resolve(
    "react-native/scripts/react_native_pods.rb",
    {paths: [process.argv[1]]},
  )', __dir__]).strip
 
  MIN_IOS_OVERRIDE = '14.0'
  if Gem::Version.new(MIN_IOS_OVERRIDE) > Gem::Version.new(min_ios_version_supported)
      min_ios_version_supported = MIN_IOS_OVERRIDE
  end  

platform :ios, min_ios_version_supported
prepare_react_native_project!
```

**Comment by brascene** [comment]
> Add the following
> 
> ```js
> require Pod::Executable.execute_command('node', ['-p',
>   'require.resolve(
>     "react-native/scripts/react_native_pods.rb",
>     {paths: [process.argv[1]]},
>   )', __dir__]).strip
>  
>   MIN_IOS_OVERRIDE = '14.0'
>   if Gem::Version.new(MIN_IOS_OVERRIDE) > Gem::Version.new(min_ios_version_supported)
>       min_ios_version_supported = MIN_IOS_OVERRIDE
>   end  
> 
> platform :ios, min_ios_version_supported
> prepare_react_native_project!
> ```

That's not resolving the issue in RCTShadowView

**Comment by terrysahaidak** [comment]
hey @brascene have you found a solution for this crash?

i just spent two days figuring out what breaks our app after upgrading to 0.74 and only after removing rive it worked...

**Comment by brascene** [comment]
> hey @brascene have you found a solution for this crash?
> 
> 
> 
> i just spent two days figuring out what breaks our app after upgrading to 0.74 and only after removing rive it worked... 

yes @terrysahaidak i found a solution,i downgraded react native version to 0.73.7 😅 although it doesn't seem a big change (0.73 and 0.74) but there actually were some major changes regarding yarn, yoga and other breaking changes.

**Comment by terrysahaidak** [comment]
@brascene it looks like downgrading rive to 7.2.0 also fixes the problem, so i'm gonna stick to it in the meantime... thanks for the response
